src/views/TabView/TabBarBottom.js

```diff
@@ -179,64 +169,84 @@ class TabBarBottom extends PureComponent<DefaultProps, Props, void> {
+            );
+            const backgroundColor = position.interpolate({
+              inputRange,
+              outputRange: (outputRange: Array<string>),
+            });
 
-          const justifyContent = this.props.showIcon ? 'flex-end' : 'center';
-          const extraProps = this._renderTestIDProps(scene) || {};
-          const { testID, accessibilityLabel } = extraProps;
+            const justifyContent = this.props.showIcon ? 'flex-end' : 'center';
+            const extraProps = this._renderTestIDProps(scene) || {};
+            const { testID, accessibilityLabel } = extraProps;
 
-          return (
-            <TouchableWithoutFeedback
-              key={route.key}
-              testID={testID}
-              accessibilityLabel={accessibilityLabel}
-              onPress={() =>
-                onPress ? onPress(scene, jumpToIndex) : jumpToIndex(index)}
-            >
-              <Animated.View
-                style={[
-                  styles.tab,
-                  isLandscape && useHorizontalTabs && styles.tabLandscape,
-                  !isLandscape && useHorizontalTabs && styles.tabPortrait,
-                  { backgroundColor },
-                  tabStyle,
-                ]}
+            return (
+              <TouchableWithoutFeedback
+                key={route.key}
+                testID={testID}
+                accessibilityLabel={accessibilityLabel}
+                onPress={() =>
+                  onPress ? onPress(scene, jumpToIndex) : jumpToIndex(index)}
               >
-                {this._renderIcon(scene)}
-                {this._renderLabel(scene)}
-              </Animated.View>
-            </TouchableWithoutFeedback>
-          );
-        })}
-      </Animated.View>
+                <Animated.View
+                  style={[
+                    styles.tab,
+                    isLandscape && useHorizontalTabs && styles.tabLandscape,
+                    !isLandscape && useHorizontalTabs && styles.tabPortrait,
+                    { backgroundColor },
+                    tabStyle,
+                  ]}
+                >
+                  {this._renderIcon(scene)}
+                  {this._renderLabel(scene)}
+                </Animated.View>
+              </TouchableWithoutFeedback>
+            );
+          })}
+        </Animated.View>
+      </SafeAreaView>
     );
   }
 }
 
 const LABEL_LEFT_MARGIN = 20;
 const LABEL_TOP_MARGIN = 15;
 const styles = StyleSheet.create({
-  tabBar: {
-    height: 49, // Default tab bar height in iOS 10+
-    flexDirection: 'row',
+  tabBarContainer: {
+    backgroundColor: '#F7F7F7', // Default background color in iOS 10
     borderTopWidth: StyleSheet.hairlineWidth,
     borderTopColor: 'rgba(0, 0, 0, .3)',
-    backgroundColor: '#F7F7F7', // Default background color in iOS 10+
+  },
+  tabBar: {
+    // height: 49, // Default tab bar height in iOS 10+
+    flexDirection: 'row',
+  },
+  tabBarLandscape: {
+    height: 29,
```

Comment:
On versions of iOS under 11 this should be 49 because the tab items will be stacked.

Reply:
Hmm, k, I'll make that change. On a related note, tab icons should also be slightly smaller on iOS 11 in landscape. Doesn't seem easy to implement that as is. @vonovak had a solution for withStatusBar that had a render prop that passed status bar height. I think a future improvement for SafeAreaView would be a render prop that passed the screen size class or something, with which you could update the tab icon size. I'll open an issue at some point.